test/modules/home/home_interactor_test.dart

```diff
@@ -0,0 +1,159 @@
+      locator.registerSingleton<SurveyRepository>(surveyRepository);
+
+      authRepository = MockAuthRepository();
+      locator.registerSingleton<AuthRepository>(authRepository);
+
+      interactor = HomeInteractorImpl();
+      interactor.delegate = delegate;
+    });
+
+    describe("it's getAuthenticatedUser is called", () {
+      final user = UserInfo();
+      beforeEach(() {
+        when(authRepository.user).thenReturn(user);
+        interactor.getAuthenticatedUser();
+      });
+
+      it("triggers delegate's authenticatedUserDidGet emits", () {
+        expect(delegate.authenticatedUserDidGet, emits(user));
+      });
+    });
+
+    describe("it's fetchSurveysFromCached is called", () {
+      context("when surveyRepository's fetchSurveysFromCached return success",
+          () {
+        final surveys = List<SurveyInfo>.empty();
+        beforeEach(() {
+          when(surveyRepository.fetchSurveysFromCached())
+              .thenAnswer((realInvocation) => Future.value(surveys));
+          interactor.fetchSurveysFromCached();
+        });
+
+        it("triggers delegate's surveysDidFetchFromCached emits", () {
+          expect(delegate.surveysDidFetchFromCached, emits(surveys));
+        });
+      });
+
+      context("when surveyRepository's fetchSurveysFromCached return failure",
+          () {
+        final exception = Exception();
+        beforeEach(() {
+          when(surveyRepository.fetchSurveysFromCached())
+              .thenAnswer((realInvocation) => Future.error(exception));
+          interactor.fetchSurveysFromCached();
+        });
+
+        it("triggers delegate's surveysDidFailToFetchFromCached emits", () {
+          expect(delegate.surveysDidFailToFetchFromCached, emits(exception));
+        });
+      });
+    });
+
+    describe("it's fetchSurveysFromRemote is called", () {
+      context("when surveyRepository's fetchSurveysFromRemote return success",
+          () {
+        context(
+            "when surveyRepository's surveysDidFailToFetchFromCached return out of update data",
+            () {
+          final cachedSurveys = [
+            SurveyInfo()..description = "cached description"
+          ];
+
+          final remoteSurveys = [SurveyInfo()..description = "description"];
+
+          beforeEach(() {
+            when(surveyRepository.fetchSurveysFromCached())
+                .thenAnswer((realInvocation) => Future.value(cachedSurveys));
+
+            when(surveyRepository.fetchSurveysFromRemote())
+                .thenAnswer((realInvocation) => Future.value(remoteSurveys));
+            interactor.fetchSurveysFromRemote();
+          });
+
+          it("triggers delegate's surveysDidFetchFromRemote emits", () {
+            expect(delegate.surveysDidFetchFromRemote,
+                emits(Tuple2(remoteSurveys, true)));
+          });
+        });
+
+        context(
+            "when surveyRepository's surveysDidFailToFetchFromCached return in of update data",
```

Comment:
I don't fully understand this sentence...can you please revise?

Reply:
@sleepylee  I mean `returns not outdated data`, but can't find a good word for it 😓